cicecore/cicedyn/analysis/ice_diagnostics.F90

```diff
@@ -1310,6 +1333,12 @@ subroutine init_mass_diags
       toten = global_sum(work1, distrb_info, field_loc_center, tarean)
       totes = global_sum(work1, distrb_info, field_loc_center, tareas)
 
+      ! north/south salt
+      call total_salt (work2)
+      totsn = global_sum(work2, distrb_info, field_loc_center, tarean)
+      totss = global_sum(work2, distrb_info, field_loc_center, tareas)
```

Comment:
Looks like you could reuse work1 here rather than add new array, work2.

Reply:
Maybe. I thought they needed to be independent.

Reply:
In this code, we do not need work2, but in the print_points code we do. The array work2 is used in a number of places, so it is not really saving us memory.

Reply:
OK, it just looks like work2 was added in init_mass_diags just for the new salt diagnostic.  work2 must exist separately in print_points and elsewhere.